### PR TITLE
[nn] add `insert` method to sequential class

### PR DESCRIPTION
**Pull request by khushi-411**
Follows #71329

cc @kshitij12345

### CODE REVIEW

**Comment by khushi-411** [comment]
Hi, gentle ping @albanD  @jbschlosser! :)

Thanks in advance!

**Comment by kshitij12345** [comment]
Hi @khushi-411,

@albanD @jbschlosser are on leave, so it might take a few days to get this PR reviewed.

File: torch/nn/modules/container.py [hunk 1]
```diff
@@ -163,6 +163,12 @@ def append(self, module: Module) -> 'Sequential':
         self.add_module(str(len(self)), module)
         return self
 
+    def insert(self, index: int, module: Module) -> 'Sequential':
+        for i in range(len(self._modules), index, -1):
```
[the remark this answers is not in the document]
Comment (kshitij12345) [reply torch/nn/modules/container.py:167]: Also should we have an assert to verify that `module` is actually an instance of `Module`?

**Comment by khushi-411** [comment]
> Hi @khushi-411,
> 
> @albanD @jbschlosser are on leave, so it might take a few days to get this PR reviewed.

Hi, @kshitij12345!

Nice, thanks for sharing with me ;)

**Review by kshitij12345 — COMMENTED** [review]
Overall looks good but I have a few inline questions.

Also, I was wondering if it should support negative indices. Python lists do support that.

```python
>>> a = [1, 2, 3]
>>> a.insert(-1, 0)
>>> a
[1, 2, 0, 3]
>>> a.insert(-2, 5)
>>> a
[1, 2, 5, 0, 3]
```

Wdyt?

File: torch/nn/modules/container.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (khushi-411) [reply torch/nn/modules/container.py:167]: I think it will be great. Will add it. Thanks for suggesting it.

**Comment by khushi-411** [comment]
Thanks for reviewing, @kshitij12345!

> Wdyt?

I think it's a pretty cool suggestion. Thanks! I've added support for negative inputs and addressed all your reviews ;)

**Comment by khushi-411** [comment]
Oops! Resolving conflicts caused many testing errors, which looks reasonable to me. I'll take a look into those. Thanks!

**Review by albanD — APPROVED** [review]
Looks good! Thanks for taking care of that!

**Pull request closed without merging**